sys/abi/src/lib.rs

```diff
@@ -546,3 +548,44 @@ impl core::convert::TryFrom<u32> for Sysnum {
         }
     }
 }
+
+#[repr(C)]
+#[derive(Default, Copy, Clone, Debug)]
+#[cfg_attr(feature = "std", derive(PackedStruct))]
+#[cfg_attr(
+    feature = "std",
+    packed_struct(size_bytes = "0x8", bit_numbering = "msb0", endian = "msb")
+)]
+pub struct SAUEntry {
+    #[cfg_attr(feature = "std", packed_field(endian = "lsb"))]
+    pub rbar: u32,
+    #[cfg_attr(feature = "std", packed_field(endian = "lsb"))]
+    pub rlar: u32,
+}
+
+pub const HEADER_MAGIC: u32 = 0x1535_6637;
+
+#[repr(C)]
+#[derive(Default)]
+#[cfg_attr(feature = "std", derive(PackedStruct))]
+#[cfg_attr(
+    feature = "std",
+    packed_struct(size_bytes = "0x80", bit_numbering = "msb0", endian = "msb")
+)]
+pub struct ImageHeader {
+    #[cfg_attr(feature = "std", packed_field(endian = "lsb"))]
+    pub magic: u32,
+    #[cfg_attr(feature = "std", packed_field(endian = "lsb"))]
+    pub entry: u32,
+    #[cfg_attr(feature = "std", packed_field(endian = "lsb"))]
+    pub vector: u32,
+    #[cfg_attr(feature = "std", packed_field(endian = "lsb"))]
+    pub total_image_len: u32,
+    #[cfg_attr(
+        feature = "std",
+        packed_field(endian = "lsb", element_size_bytes = "0x8")
+    )]
+    pub sau_entries: [SAUEntry; 8],
+    // pad for VTOR alignment (128)
```

Comment:
I'd suggest letting the linker script manage the padding, particularly because different implementations have different VTOR alignment requirements.